rasa/core/nlg/template.py

```diff
@@ -104,15 +104,19 @@ def _fill_template_text(
         template_vars = self._template_variables(filled_slots, kwargs)
 
         # Filling the template variables in the template text
-        if template_vars and "text" in template:
+        if template_vars and ("text" in template or "image" in template):
             try:
                 # transforming template tags from
                 # "{tag_name}" to "{0[tag_name]}"
                 # as described here:
                 # https://stackoverflow.com/questions/7934620/python-dots-in-the-name-of-variable-in-a-format-string#comment9695339_7934969
                 # assuming that slot_name do not contain newline character here
-                text = re.sub(r"{([^\n]+?)}", r"{0[\1]}", template["text"])
-                template["text"] = text.format(template_vars)
+                if "text" in template:
```

Comment:
can reduce the duplication here a bit? E.g. sth like

```python
for key in ["text", "image"]:
  if key in template:
    image = re.sub(r"{([^\n]+?)}", r"{0[\1]}", template[key])
    template[key] = image.format(template_vars)
```

Reply:
awesome 👍

Reply:
Thanks a lot @wochinge :)